usagov/vote-gov-drupal
# Vote 801 -> cypress installation

link to ticket -> https://bixal-projects.atlassian.net/browse/VOTE-801

### Purpose of PR: 
This pr will add cypress to vote.gov drupal repo 

### Extra Info on work:

- I created a `testing` folder... in order to move the `cypress.config.js`, `package.json` and `package-lock.json` into the cypress folder it needed an extra one on top of it in order for the tests to work.  I was not able to just add them to the cypress folder and everything still works 
- there is a sample spec file to show that everything is working.. this just simply loads the page and checks that it has loaded, once there is more migrated in this will be removed.
- there will be future configuration needed such as scripts and other tests added but will do that on an as-needed basis. 

### How to test:

1. pull down branch 
2. start local env 
3. cd into `testing` folder 
4. `npm run cy:test` this will run test headless 
5. `npm run cy:open` this will open the cypress runner 

for the last 2 steps ensure that the test is passing as expected
